src/test-database-manager.ts

```diff
@@ -0,0 +1,142 @@
+import { randomUUID } from 'crypto'
+import { Client } from 'pg'
+import { DataSource, Logger } from 'typeorm'
+import { PostgresConnectionOptions } from 'typeorm/driver/postgres/PostgresConnectionOptions'
+
+const noop = () => void 0
+const errorOnlyLogger: Logger = {
+  logQueryError(error: string | Error, query: string, parameters?: any[]) {
+    // eslint-disable-next-line no-console
+    console.error('TYPEORM:ERROR', error, query, parameters)
+  },
+  logQuery: noop,
+  logQuerySlow: noop,
+  logSchemaBuild: noop,
+  logMigration: noop,
+  log: noop,
+}
+
+export class TestDatabaseManager {
```

Comment:
Hi @cuzzlor, thanks for the suggestion, today I learnt about `AsyncDisposable` interface. 

I tried to implement it in this library but faced an issue:
- `AsyncDisposable` requires the resource to be declared with `using`. I can't find any resources on how to fit `using` into Jest setup and teardown.

Do you know a way to address it?

Reply:
Can you just create it in the test setup and call async dispose in the test teardown without using a `using` ?

Feel free to ignore my suggestions, I was thinking it might be useful to explore the layout of the new disposable features and align towards if it's not too clunky.